### PR TITLE
Update kube-openapi and gnostic dependencies

### PR DESCRIPTION
**Pull request by howardjohn**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This will be required to pull in the changes in https://github.com/kubernetes/kube-openapi/pull/220. Once that PR is merged, I will update the go.mod to pull in the update.

#### Which issue(s) this PR fixes:
None

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/client-go/discovery/discovery_client_test.go [hunk 1]
```diff
@@ -547,8 +547,8 @@ func TestGetOpenAPISchema(t *testing.T) {
 	if err != nil {
 		t.Fatalf("unexpected error getting openapi: %v", err)
 	}
-	if e, a := returnedOpenAPI(), got; !reflect.DeepEqual(e, a) {
-		t.Errorf("expected %v, got %v", e, a)
+	if e, a := returnedOpenAPI(), got; !golangproto.Equal(e, a) {
```
Comment (yliaog) [inline staging/src/k8s.io/client-go/discovery/discovery_client_test.go:550]: why is this needed?

Comment (howardjohn) [reply staging/src/k8s.io/client-go/discovery/discovery_client_test.go:550]: Its not legal to do reflect.DeepEqual on protos generated by newer protoc versions. https://groups.google.com/g/golang-nuts/c/-p-KiwYE598 describes this a bit.

The issue is data gets encoded into private fields that are implementation details and should not be directly compared. The recommended way is actually using protocmp but I didn't want to introduce a new dependency

Comment (yliaog) [reply staging/src/k8s.io/client-go/discovery/discovery_client_test.go:550]: ok. why adding '\n' in the line below?

Comment (howardjohn) [reply staging/src/k8s.io/client-go/discovery/discovery_client_test.go:550]: oh that was to make it easier to view the failure, I can revert if necessary

Comment (yliaog) [reply staging/src/k8s.io/client-go/discovery/discovery_client_test.go:550]: that is fine as is. just wondering that '\n' is only added in this one line below, but not in others.

**Comment by christopherhein** [comment]
@howardjohn thank you for doing all this, really helpful glad to see this being done. Sounds like it might not land until 1.21 any suggestions for how we could use this with CR as of 0.8.X given that has 1.20 deps? I see CR has 1.21 in the 0.9.X alphas.

**Comment by howardjohn** [comment]
@christopherhein you should be able to just use the new kube-openapi directly? unless you import kubernetes/kubernetes directly, then it may cause issues. For us it didn't, and we were using the new kube-openapi (via a fork) for a while now

**Comment by christopherhein** [comment]
> @christopherhein you should be able to just use the new kube-openapi directly? unless you import kubernetes/kubernetes directly, then it may cause issues. For us it didn't, and we were using the new kube-openapi (via a fork) for a while now

Yea, my issue is where I'm importing k/k even though it's generally frowned upon :/

File: staging/src/k8s.io/client-go/discovery/discovery_client_test.go (hunk 1 above)
Comment (liggitt) [reply staging/src/k8s.io/client-go/discovery/discovery_client_test.go:550]: > Its not legal to do reflect.DeepEqual on protos generated by newer protoc versions. 

I see changes to paths the might be hit by openapi serving code, which cares about equality (https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/controller.go#L211-L213)

@sttts, do we have tests in place that would catch spurious detection of deltas

**Comment by liggitt** [comment]
one question about the DeepEqual change, lgtm otherwise

**Comment by gautierdelorme** [comment]
/retest

I'm not sure if this test is supposed to be flaky but it fails with `timed out waiting for the condition` since a few hours...

**Comment by gautierdelorme** [comment]
/retest

opened #101275 for further investigation about failing tests
